sparx/bin/sxctf_refine.py

```diff
@@ -0,0 +1,890 @@
+#! /usr/bin/env python
+"""
+CTF Refinement with error assessment for SPHIRE
+
+#
+# Author: Thorsten Wagner 02/18/2019 (thorsten.wagner@mpi-dortmund.mpg.de)
+# Copyright (C) 2019 Max planck institute for molecular physiology, Dortmund
+#
+# This software is issued under a joint BSD/GNU license. You may use the
+# source code in this file under either license. However, note that the
+# complete EMAN2 and SPARX software packages have some GPL dependencies,
+# so you are responsible for compliance with the licenses of these packages
+# if you opt to use BSD licensing. The warranty disclaimer below holds
+# in either instance.
+#
+# This complete copyright notice must be included in any revised version of the
+# source code. Additional authorship citations may be added, but existing
+# author citations must be preserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307 USA
+#
+#
+"""
+# pylint: disable=C0330
+import argparse
+import statistics
+import time
+import multiprocessing
+import os
+import itertools
+import numpy as np
+
+from tqdm import tqdm
+from scipy import ndimage
+
+import sxctf_refine_plotting
+import sxctf_refine_io
+import filter as fltr
+import EMAN2
+import global_def
+from projection import prgl
+
+
+EXAMPLE_TEXT_MERIDIEN = """Example:
+
+sxrefinectf.py -sxm /path/to/meridien/refinement/folder/ -s bdb:/path/to/stack/STACKNAME 
+              -o bdb:/your/output/path/bdb_file_name -os /statistics/out/path/ 
+              -m /pth/adp_mask/mask.hdf -r 0.2 -d 0.01 -res 2.2 -apix 1.07
+
+"""
+
+EXAMPLE_TEXT_MANUAL = """Example:
+
+sxrefinectf.py -v1 /path/to/volume/vol_0_unfil_XXX.hdf -v2 /path/to/volume/vol_0_unfil_XXX.hdf 
+            -c /path/to/meridien/mainXXX/chunk_0_XXX.txt -s bdb:/path/to/stack/STACKNAME 
+            -p /path/to/merdien/refinement/final_params_XXX.txt -o bdb:/your/output/path/bdb_file_name 
+            -os /statistics/out/path/ -m /pth/adp_mask/mask.hdf -r 0.2 -d 0.01 -res 2.2 -apix 1.07
+
+"""
+
+ARGPARSER = argparse.ArgumentParser(
```

Comment:
Can we put the argparser part in a function calles something like "parse_args()" to avoid global variables?

Parse the args before the main function, so that main can be called from within a script.